.eslintrc.js

```diff
@@ -400,10 +400,185 @@ module.exports = {
+        'no-sparse-arrays': 'error',
+        'no-this-before-super': 'error',
+        // This will be turned on after bug fixes are mostly complete
+        // 'no-undef': 'warn',
+        'no-unreachable': 'error',
+        'no-unsafe-finally': 'error',
+        'no-useless-call': 'error',
+        // This will be turned on after bug fixes are mostly complete
+        // 'no-useless-catch': 'warn',
+        'no-useless-concat': 'error',
+        'no-useless-computed-key': 'error',
+        // This will be turned on after bug fixes are mostly complete
+        // 'no-useless-escape': 'warn',
+        'no-useless-rename': 'error',
+        // This will be turned on after bug fixes are mostly complete
+        // 'no-useless-return': 'warn',
+        // This will be turned on after bug fixers are mostly complete
+        // 'no-void': 'warn',
+        'one-var-declaration-per-line': 'error',
+        'prefer-object-spread': 'error',
+        'prefer-promise-reject-errors': 'error',
+        'prefer-rest-params': 'error',
+        'prefer-spread': 'error',
+        // This style will be turned on after most bugs are fixed
+        // 'prefer-template': 'warn',
+        // This style will be turned on after most bugs are fixed
+        // quotes: ['warn', 'single', { avoidEscape: true }],
+        'react/boolean-prop-naming': 'error',
+        'react/button-has-type': 'error',
+        'react/forbid-dom-props': 'error',
+        // This will go from warn to error when this is fixed:
+        // https://github.com/elastic/ingest-dev/issues/468
+        'react/no-access-state-in-setstate': 'warn',
+        // This style will be turned on after most bugs are fixed
+        // 'react/no-children-prop': 'warn',
+        'react/no-danger-with-children': 'error',
+        'react/no-deprecated': 'error',
+        'react/no-did-mount-set-state': 'error',
+        // This will go from warn to error when this is fixed:
+        // https://github.com/elastic/ingest-dev/issues/468
+        'react/no-did-update-set-state': 'warn',
+        'react/no-direct-mutation-state': 'error',
+        'react/no-find-dom-node': 'error',
+        'react/no-redundant-should-component-update': 'error',
+        'react/no-render-return-value': 'error',
+        'react/no-typos': 'error',
+        'react/no-string-refs': 'error',
+        'react/no-this-in-sfc': 'error',
+        // This can go from warn to error once this is fixed
+        // https://github.com/elastic/ingest-dev/issues/467
+        'react/no-unescaped-entities': 'warn',
+        'react/no-unsafe': 'error',
+        'react/no-unused-prop-types': 'error',
+        'react/no-unused-state': 'error',
+        // will introduced after the other warns are fixed
+        // 'react/sort-comp': 'error',
+        'react/void-dom-elements-no-children': 'error',
+        'react/jsx-boolean-value': ['error', 'warn'],
+        // will introduced after the other warns are fixed
+        // 'react/jsx-no-bind': 'error',
```

Comment:
I think there's a strong argument for getting [this rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md) on sooner than latter from a perf standpoint. Should we open an issue for tracking to raise visibility? Turning it on showed `120 errors` and it's pretty rampant around the DnD areas within the app.